drivers/src/sha2_512_384.rs

```diff
@@ -418,70 +453,121 @@ impl<'a, const HASH_SIZE: u16> Sha2DigestOp<'a, HASH_SIZE> {
     }
 }
 
-impl<'a> Sha2DigestOp<'a, 384> {
+impl<'a> Sha2DigestOpTrait<'a, Sha384> for Sha2DigestOp<'a, Sha384> {
+    fn as_digest_op(&mut self) -> &mut Sha2DigestOp<'a, Sha384> {
+        self
+    }
+}
+
+impl<'a> Sha2DigestOpTrait<'a, Sha512> for Sha2DigestOp<'a, Sha512> {
+    fn as_digest_op(&mut self) -> &mut Sha2DigestOp<'a, Sha512> {
+        self
+    }
+}
+
+/// Trait for SHA-2 digest operations
+pub trait Sha2DigestOpTrait<'a, V: Sha2Variant>: Sized {
```

Comment:
I'm not 100% sure why we need this trait. Why not implement this method directly on the generic struct, and not have to boomerang back and forth through the `OpTrait` and the `Op`?

Reply:
I think I understood yourfeedback and I removed that Sha2Variant trait.